Firestore/CMakeLists.txt

```diff
@@ -15,6 +15,53 @@
 cmake_minimum_required(VERSION 2.8.11)
 project(firestore C CXX)
 
+option(ASAN "Build with Address sanitizer (mutually exculsive with TSAN)" OFF)
+# TODO(varconst): msan
+# Memory sanitizer is more complicated:
+# - it requires all dependencies to be compiled with msan enabled (see
+#   https://github.com/google/sanitizers/wiki/MemorySanitizerLibcxxHowTo);
+# - AppleClang doesn't support it.
+option(TSAN "Build with Thread sanitizer (mutually exculsive with ASAN)" OFF)
```

Comment:
Conventionally these kinds of options are `WITH_FOO`. For example, see other CMake files that do it:: https://github.com/facebook/rocksdb/blob/master/CMakeLists.txt#L258. This is not universal though, so feel free to ignore.

Reply:
Done. I like that this prefix makes them grouped together in the cache.